lib/pbench/test/unit/agent/task/test_results_push.py

```diff
@@ -21,10 +22,14 @@ class TestResultsPush:
     META_SWITCH = "--metadata"
     META_TEXT_FOO = "pbench.sat:FOO"
     META_TEXT_TEST = "pbench.test:TEST"
+    RELAY_SWITCH = "--relay"
+    SRVR_SWITCH = "--server"
+    RELAY_TEXT = "http://relay.example.com"
+    SRVR_TEXT = "http://pbench.test.example.com"
     URL = "http://pbench.example.com/api/v1"
 
     @staticmethod
-    def add_http_mock_response(
+    def add_server_mock_response(
```

Comment:
I gather that you changed this name to make it stand better in opposition to the new `relay_mock()`...but why did you make that a fixture and leave this as a helper function?  I would have thought that it would be better to have a consistent approach for both of them.

Reply:
I avoided changing the usage here, even though the over-generality of the name bothered me, while in the relay case I was starting from scratch.

There's actually a bigger problem here, because *both* take arguments which are never used and, in the case of the relay, can't be used because we can't pass arguments to a fixture that way. And the test cases here aren't constructed to require that flexibility, so the arguments are pointless and both can be trimmed down.

Ideally the API error cases would be tested here, but the `move` test is much more thorough and on this long hot muggy day I find it difficult to feel motivated to improve these tests. 🌞

Reply:
> both take arguments which are never used

I'm not sure what you mean.  The parameters are referenced inside the function bodies, and arguments are supplied to (now) `add_server_mock_response()` in `test_push_status()` at [line 298](https://github.com/distributed-system-analysis/pbench/blob/e208af4a64275c7a53f1f7242e5816fc77d316d4/lib/pbench/test/unit/agent/task/test_results_push.py#L298).

> in the case of the relay, can't be used because we can't pass arguments to a fixture that way

Right, but that's easily finessed:  you have the fixture return a function as its value and that function accepts the arguments...but, I think that works out to being equivalent to making `relay_mock()` into a simple helper function.

> this long hot muggy day

Oh, Dave...it's not even July, yet....  (My kids, who, unlike me, have actually been outside today, claim that it's a much nicer day than yesterday....)  Perhaps it is time for you to turn on your A/C.  😞

Reply:
Yeah, I missed the *one* test case that actually supplied non-default arguments. Sigh.